universal_transfer_operator/pyproject.toml

```diff
@@ -0,0 +1,149 @@
+
+[project]
+name = "universal-transfer-operator"
+dynamic = ["version"]
+description = """
+"""
+
+authors = [
+    { name = "Astronomer", email = "humans@astronomer.io" },
+]
+readme = "README.md"
+license = { file = "LICENSE" }
+
+requires-python = ">=3.7"
+dependencies = [
+    "apache-airflow>=2.0",
+    "attrs>=20.0",
+    "pandas>=1.3.4,<2.0.0", # Pinning it to <2.0.0 to avoid breaking changes
+    "pyarrow",
+    "python-frontmatter",
+    "smart-open",
+    "SQLAlchemy>=1.3.18",
+    "apache-airflow-providers-common-sql"
+]
+
+keywords = ["airflow", "provider", "astronomer", "sql", "decorator", "task flow", "elt", "etl", "dag"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Database",
+    "Framework :: Apache Airflow",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+]
+
+[project.optional-dependencies]
+tests = [
+    "pytest>=6.0",
+    "pytest-split",
+    "pytest-dotenv",
+    "requests-mock",
+    "pytest-cov",
+    "pytest-describe",
+    "types-requests",
+    "mypy",
+    "sqlalchemy-stubs", # Change when sqlalchemy is upgraded https://docs.sqlalchemy.org/en/14/orm/extensions/mypy.html
+]
+google = [
+    "protobuf<=3.20", # Google bigquery client require protobuf <= 3.20.0. We can remove the limitation when this limitation is removed
+    "apache-airflow-providers-google>=6.4.0",
+    "sqlalchemy-bigquery>=1.3.0",
+    "smart-open[gcs]>=5.2.1"
+]
+snowflake = [
+    "apache-airflow-providers-snowflake",
+    "snowflake-sqlalchemy>=1.2.0",
+    "snowflake-connector-python[pandas]",
+]
+postgres = [
+    "apache-airflow-providers-postgres",
+]
+amazon = [
+    "apache-airflow-providers-amazon>=5.0.0",
+    "s3fs",
+    "smart-open[s3]>=5.2.1",
+]
+azure = [
+    "apache-airflow-providers-microsoft-azure",
+    "azure-storage-blob",
+    "smart-open[azure]>=5.2.1",
+]
+openlineage = ["openlineage-airflow>=0.17.0"]
+
+fivetran = ["airflow-provider-fivetran>=1.1.3"]
+
+databricks = ["databricks-cli",
+    "apache-airflow-providers-databricks"]
+
+all = [
+    "apache-airflow-providers-amazon",
+    "apache-airflow-providers-google>=6.4.0",
+    "apache-airflow-providers-postgres",
+    "apache-airflow-providers-snowflake",
+    "smart-open[all]>=5.2.1",
+    "snowflake-connector-python[pandas]",
+    "snowflake-sqlalchemy>=1.2.0",
+    "sqlalchemy-bigquery>=1.3.0",
+    "databricks-cli",
+    "apache-airflow-providers-databricks",
+    "s3fs",
+    "protobuf<=3.20", # Google bigquery client require protobuf <= 3.20.0. We can remove the limitation when this limitation is removed
+    "openlineage-airflow>=0.17.0",
+    "airflow-provider-fivetran>=1.1.3",
+    "apache-airflow-providers-microsoft-azure",
+    "azure-storage-blob"
+]
```

Comment:
can you take a pass at all the deps please -- and add only ones that are used by UTO in this PR

Reply:
Sure. Doing it now

Reply:
[Done](https://github.com/astronomer/astro-sdk/pull/1492/commits/c2e08eaec89c463d8e83e7a8fa523231e56ede4f)